### DISCUSSION TITLE
Package files that are not UTF-8 encoded are silently ignored

### DISCUSSION THREAD

**Discussion by pritishkapoor** [renovatebot/renovate#26307] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.93.1

### If you're self-hosting Renovate, select which platform you are using.

Azure DevOps (dev.azure.com)

### Was this something which used to work for you, and then stopped?

I have not seen this working

### Describe the problem

Renovate detects the package files (E.g. requirements.txt) but does not log the reason for ignoring the ones that are not UTF-8 encoded.

Minimal reproduction repository: https://github.com/ridhimax/renovate-utf8-issue

### Relevant debug logs

<details><summary>Logs</summary>

```

DEBUG: Matched 1 file(s) for manager pip_requirements: requirements.txt (repository=my-project/my-repo)
DEBUG: manager extract durations (ms) (repository=my-project/my-repo)
       "managers": {"pip_requirements": 3}
DEBUG: Found 0 package file(s) (repository=my-project/my-repo)
 INFO: Repository has no package files - skipping (repository=my-project/my-repo)
DEBUG: Repository result: no-package-files, status: disabled, enabled: false, onboarded: undefined (repository=my-project/my-repo)

```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment]
I'm not sure of how to achieve this but you're welcome to submit a PR if you do.

Debugging it, the readLocalFile() call with utf8 does return content:

<img width="912" alt="image" src="https://github.com/renovatebot/renovate/assets/6311784/faa9471a-b916-47ed-8e63-042b33de1033">

But then the content is not _parsed_ because of the special characters.

Maybe we'd need to parse it as a Buffer first, then use some of the suggestions like in https://stackoverflow.com/questions/50045841/how-to-detect-file-encoding-in-nodejs before then returning it as text

**Comment by pritishkapoor** [comment]
Thanks for looking into it. I read about encoding detection - the problem with it is that it does not guarantee correct detection, especially for small files.

What is in 'res' though (for a non-UTF8)? Is this where the packageFile got filtered, or the filtering happens at a later point? If this is where it got filtered, then would an 'else' condition help here? With a DEBUG message like "Skipping ${packageFile} since it could not be parsed, possibly because the file does not use UTF-8 encoding"? We could even add to Renovate docs that UTF-8 encoding is mandatory for the package files to be detected by Renovate.

```
if (res) {
  packageFiles.push({
    ...res,
    packageFile,
  });
}
```

**Reply by rarkins** [reply, marked as answer]
You're welcome to submit PRs for:
- Making it clearer in docs that utf8 are required
- Adding logs when a packageFile fails to parse
